xsc_test.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-cli-core/v2/common/format"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-cli-security/formats"
+	"github.com/jfrog/jfrog-cli-security/scangraph"
+	securityTests "github.com/jfrog/jfrog-cli-security/tests"
+	securityTestUtils "github.com/jfrog/jfrog-cli-security/tests/utils"
+	"github.com/jfrog/jfrog-cli-security/utils"
+	"github.com/jfrog/jfrog-client-go/utils/tests"
+	"github.com/jfrog/jfrog-client-go/xsc/services"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestXscAnalyticsForAudit(t *testing.T) {
+	securityTestUtils.InitSecurityTest(t, scangraph.GraphScanMinXrayVersion, services.AnalyticsMetricsMinXscVersion)
+	reportUsageCallBack := tests.SetEnvWithCallbackAndAssert(t, coreutils.ReportUsage, "true")
+	defer reportUsageCallBack()
+	// Scan npm project and verify that analytics general event were sent to XSC.
+	output := testXrayAuditNpm(t, string(format.SimpleJson))
+	validateAnalyticsBasicEvent(t, output)
+}
+
+func validateAnalyticsBasicEvent(t *testing.T, output string) {
+	// Get MSI.
+	var results formats.SimpleJsonResults
+	err := json.Unmarshal([]byte(output), &results)
+	assert.NoError(t, err)
+
+	// Verify analytics metrics.
+	am := utils.NewAnalyticsMetricsService(securityTests.XscDetails)
+	assert.NotNil(t, am)
+	event, err := am.GetGeneralEvent(results.MultiScanId)
+	assert.NoError(t, err)
+
+	// Event creation and addition information.
+	assert.Equal(t, "cli", event.Product)
+	assert.Equal(t, 1, event.EventType)
+	assert.NotEmpty(t, event.EventStatus)
+	assert.NotEmpty(t, event.AnalyzerManagerVersion)
+	// The information that was added after updating the event with the scan's results.
+	assert.NotEmpty(t, event.TotalScanDuration)
+	assert.True(t, event.TotalFindings > 0)
+}
```

Comment:
No need for specific file for analytics integration and add another scan for that. it can be added to an existing test at `audit_test.go` for example `TestXrayAuditJasSimpleJson` since it will also have JAS issues. the func `validateAnalyticsBasicEvent` can be moved to `tests/utils/test_validation.go`

Reply:
again as I mentioned before - I think that it's necessary to have a separate section for XSC testing and to have a test that tests analytics and nothing else.
I don't think that adding this in the audit section in one random test and validating analytics after other things will be right and clear.

Reply:
and I will add here also about "InitSecurityTest" - we are going to work mostly with XSC instead of xray for now on - so now when the infrastructure is still small it's time to add testing xsc support.